src/polycap-out.c

```diff
@@ -0,0 +1,51 @@
+#include <string.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include "hdf5.h"
```

Comment:
@tschoonj Compilatie failt op de hdf5.h header inclusion. Wil jij hier eens naar kijken? Philip heeft het proberen installeren op de server, en lijkt gelukt (`/usr/local/hdf5/`) maar bij compilatie vindt de code nog steeds de header niet.
Kvermoed dat ik iets moet aanpassen in de automake files, maar kweet niet goed wat. Deels denk ik in `configure.ac` best controleren of de package aanwezig is, maar vast ook flags in `src/Makefile.am` maar dna geen idee dewelke.

Reply:
Ik heb zonet hdf5-devel op uw `xmi4` lokaal geïnstalleerd.

Kan je nog eens proberen te compileren? Ik denk dat hij op zijn minst de headers zou moeten vinden nu aangezien die in `/usr/include` zitten. Als het nog altijd niet lukt, voeg dan `CPPFLAGS=-I/usr/include` toe aan uw `configure` commando.

Zelf als de header dan gevonden wordt, en er wordt succesvol gecompileerd (wat ik betwijfel want ik zie al minstens 1 fout in de code :wink) zal de linker nog altijd klagen. 
Voeg daarvoor toe aan `configure`: `LIBS=-L/usr/lib64 -lhdf5`

Reply:
Normaal moet dit inderdaad via configure automatisch gebeuren: ik zal daarvoor bij gelegenheid de nodige code schrijven. Waarschijnlijk volgende week.

Reply:
Oke, nu vindt hij zeker en vast de header :)

>want ik zie al minstens 1 fout in de code :wink
Thou dost tease... :P 

Nu foutjes eruit halen en zien of ik dan goed kan linken :)

Reply:
En voor mijn persoonlijke scholing... Hoe doe je dan die installatie van zo'n header? Want ksnap niet goed waarom hetgeen Philip probeerde niet lukte. (we volgden immers gewoon de stapjes uit de installation manual)

Reply:
Oh en de `LIBS` moeten meegegeven worden met dubbele apostroffen:

`LIBS="-L/usr/lib64 -lhdf5"`

Reply:
Op CentOS en de meeste andere Linux distributies wordt er een strikte scheiding gevolgd van een libraries en de headers. Als je immers enkel een bepaald software pakket will installeren met de package manager downloadt je de binaries ervan, dus al gecompileerd, en dus geen headers nodig van de libraries ervan. Enkel als je wil gaan developen met een bepaalde library heb je de headers ervan nodig.

Er was dus als een pakket `hdf5` geïnstalleerd op uw machine, de headers heb ik daarstraks geïnstalleerd als `hdf5-devel`. In de regel bevatten `-devel` packages dus de headers (evenals static libraries en bijvoorbeeld pkg-config files maar dit terzijde). Deze `-devel` naam conventie wordt gevolgd op CentOS, RHEL, Fedora en andere terwijl distributies uit de Debian/Ubuntu familie eerder `-dev` volgen.

In het vervolg dus, als je headers mist, zoek dan eerst of er geen `-devel` pakket van bestaat.

Reply:
Na toevoegen van `LIBS=-L/usr/lib64 -lhdf5` aan `configure` krijg ik volgende melding bij runnen van ../configure (in build)

`../configure: line 13181: -lhdf5: command not found` en nog eens hetzelfde op lijn 15556

Reply:
Zie mijn eerdere [post](https://github.com/PieterTack/polycap/pull/9#discussion_r169982973)

Reply:
Nog altijd geen volledig succes. De fout bij `../configure` run is nu wel weg (bleek overigens dat die lijn van LIBS=... er al vanzelf was ingezet, daar het met mijn toevoeging er twee keer in stond, vandaar ook de fout op twee lijnen).
Echter wanneer ik nu make uitvoer in build klaagt hij nog steeds over de link.
Khad dan nog eens autoreconf uitgevoerd, maar dan was de lijn verdwenen uit `configure`.

Reply:
Heeft het mss iets te maken met https://stackoverflow.com/questions/37007495/caffe-didnt-see-hdf5-h-when-compiling/37057313#37057313 ?

Reply:
Kan je me eens je volledige configure commando geven?

Nog altijd linker probleem? Wat is de exacte output plus link commando waar hij faalt?

Na configure te draaien doe je best altijd eens `make clean`, gevolgd door `make`, om zeker te zijn...

Reply:
Configure commando (in build/): `../configure CFLAGS="-g -O0 -Wall"`

Na make clean nog altijd probleem dat hij de references naar de H5 functies niet kent.
`libtool: link: gcc -fopenmp -g -o0 -Wall -fopenmp -o .libs/polycap polycap-main.o ./.libs/libpolycap.so -L/usr/local/lib -lgsl -lgslcblas /usr/local/lib/libxrl.so -lm -Wl,/home/ptack/poly_raytrace/build/src/.libs`

Reply:
Oke, nu geen probleem meer. Wss beetje misverstand eerder.
Heb de `LIBS="-L..."` toegevoegd aan `configure.ac` ipv configure. Als ik dan make clean, ../configure en make run werkt het wel :)

Reply:
Je moet je `LIBS` meegeven aan configure eh

Probeer eens:

```bash
../configure CFLAGS="-g -O0 -Wall" LIBS="-L/usr/lib64 -lhdf5"
make clean
make
```